src/main/webapp/assets/js/feedScript.js

```diff
@@ -921,3 +994,94 @@ function toggleLegend() {
     legend.style.display = 'none';
   }
 }
+
+/**
+ * Shows/Hides the filters to the user.
+ */
+function toggleFilters() {
+  const filterCard = document.getElementById('filter-card');
+
+  if (getComputedStyle(filterCard, null).display === 'none') {
+    filterCard.style.display = 'block';
+    toggleFiltersButton.style.backgroundColor = 'white';
+    toggleFiltersButton.style.color = 'var(--dark-maroon)';
+    toggleFiltersButton.style.border = '1px solid var(--dark-maroon)';
+  } else {
+    filterCard.style.display = 'none';
+    // Using '' makes the colors fall back to CSS defaults
+    toggleFiltersButton.style.backgroundColor = '';
+    toggleFiltersButton.style.color = '';
+    toggleFiltersButton.style.border = '';
+  }
+}
+
+/**
+ * Filters the posts the user sees on the page based
+ * on the current state of the filter inputs.
+ */
+function filterAndUpdatePagePosts() {
+  // Grab the filter inputs
+  const filters = {
+    numPeople: parseInt(document.getElementById('num-people').value),
+    happeningNow: document.getElementById('happening-now').checked,
+    distance: parseFloat(document.getElementById('distance').value),
+    keywords: document.getElementById('keywords').value === '' ?
+      [] : document.getElementById('keywords').value.split(',').map((e) => e.trim()),
+  };
+
+  const location = userLocation ?
+    {lat: userLocation.coords.latitude, long: userLocation.coords.longitude} :
+    null;
+
+  // Perform the filtering on the source of the page's posts
+  const filteredPosts = filterPosts(posts, new Date(), location, filters);
+
+  // Update the posts shown on the page
+  removePosts();
+  removeMarkers();
+  addPosts(filteredPosts);
+  addMarkers(filteredPosts);
+}
+
+/**
+ * Applies the filtering algorithm to an array of posts.
+ * @param {Array<PostInfo>} posts - The posts to filter.
+ * @param {Date} now - The current datetime.
+ * @param {Object} userLocation - The user's location (lat/long).
+ * @param {Filter} filters - The parameters by which to filter.
+ * @return {Array <PostInfo>} - The filtered list of posts.
+ */
+function filterPosts(posts, now, userLocation, filters) {
+  let filteredPosts = posts;
+  if (filters.numPeople !== NUM_PEOPLE_FILTER_DEFAULT) {
+    filteredPosts = filteredPosts.filter((p) => p.numOfPeopleFoodWillFeed >= filters.numPeople);
+  }
+  if (filters.happeningNow !== HAPPENING_NOW_FILTER_DEFAULT) {
+    filteredPosts = filteredPosts.filter((p) => {
+      return now >= p.eventStartTime && now <= p.eventEndTime;
+    });
+  }
+  if (filters.distance !== DISTANCE_FILTER_DEFAULT && userLocation) {
+    filteredPosts = filteredPosts.filter((p) => {
```

Comment:
Extract the math here into its own method and add some comments on what this calculation is doing.

Reply:
Ok!